test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -398,23 +398,17 @@ void testUtil(const TestUtilOptions& options) {
                   server_connection->ssl()->subjectLocalCertificate());
       }
       if (!options.expectedPeerCert().empty()) {
-        std::string urlencoded =
-            absl::StrReplaceAll(options.expectedPeerCert(), {{TestEnvironment::newLine, "%0A"},
-                                                             {" ", "%20"},
-                                                             {"+", "%2B"},
-                                                             {"/", "%2F"},
-                                                             {"=", "%3D"}});
+        std::string urlencoded = absl::StrReplaceAll(
```

Comment:
I think it would be better to split the test and build changes where possible, upstream these tests changes in a dedicated PR.

Reply:
There are a ridiculously small number of source changes here, which are all easily understood from the primary commit context. If you insist, I'll split it.

Reply:
I disagree with the definition of ridiculously small ;-) I'm not going to insist, but as a reviewer I think it makes life easier to split these concerns usually.

Reply:
I've commented above on the one //source/... case we should consider peeling away from this patch, everything else is test or build.

Reply:
There are no build changes that don't depend on the test changes, and vis-a-versa. So there would be no means to pass CI, so I'm really unclear on your ask.